addons/gamification/tests/test_challenge.py

```diff
@@ -184,6 +184,62 @@ def test_30_create_challenge_with_sum_goal(self):
             "Challenge failed to start",
         )
 
+    def test_40_create_challenge_with_sum_goal(self):
+        challenge = self.env['gamification.challenge'].create({
+            'name': 'test',
+            'state': 'draft',
+            'user_domain': '[("active", "=", True)]',
+            'reward_id': 1,
+        })
+
+        Goals = self.env['gamification.goal']
+
+        model = self.env['ir.model'].search([('model', '=', 'res.company')])[0]
+        field = self.env['ir.model.fields'].search([('model', '=', 'res.company'), ('name', '=', 'active')])[0]
+
+        sum_goal = self.env['gamification.goal.definition'].create({
+            'name': 'test',
+            'computation_mode': 'sum',
+            'model_id': model.id,
+            'field_id': field.id
+        })
+
+        self.env['gamification.challenge.line'].create({
+            'challenge_id': challenge.id,
+            'definition_id': sum_goal.id,
+            'condition': 'higher',
+            'target_goal': 1
+        })
+
+        challenge.action_start()
+
+        self.assertEqual(
+            challenge.state,
+            'inprogress',
+            "Challenge failed to start",
+        )
+
+        goal_ids = Goals.search([('challenge_id', '=', challenge.id), ('state', '!=', 'draft')])
+        self.assertEqual(len(goal_ids), len(challenge.line_ids) * len(challenge.user_ids.ids), "Incorrect number of goals generated, should be 1 goal per user, per challenge line")
+
+        demo = self.user_demo
+        demo.tz = "Europe/Brussels"
```

Comment:
This can only work if your definition is about `res.users`.`tz` as in the `gamification.goal.definition` `definition_base_timezone`...